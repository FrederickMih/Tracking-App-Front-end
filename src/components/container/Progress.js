import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import Header from './Header';
import {
  getMeasuresStatus,
  getProgressReport,
  getTotalData,
  loadMeasures,
} from '../../Redux/slicers/measure';
import ProgressCircle from './ProgressCircle';
import { TARGET } from '../../useful-info/constants';
import '../../styles/Progress.css';

const Progress = () => {
  const title = 'Progress Report';
  const dispatch = useDispatch();
  const totalData = useSelector(getTotalData);
  const progressReport = useSelector(getProgressReport);
  const progressPercent = (totalData / TARGET) * 100;
  const measures = useSelector(getMeasuresStatus);

  useEffect(async () => {
    await dispatch(loadMeasures());
  }, []);

  if (measures.status === 401) {
    return <Redirect to="/home" />;
  }

  return (
    <>
      <Header title={title} />
      <div className="progress-stats">
        <div>
          <ProgressCircle value={progressPercent} color="primary" />
          <div className="progress-stats-label">Achieved</div>
        </div>
        <div>
          <ProgressCircle
            value={100 - progressPercent}
            color={progressPercent >= 100 ? 'primary' : 'secondary'}
          />
          <div className="progress-stats-label">Lag</div>
        </div>
      </div>
      <div className="progress-items">
        {Object.entries(progressReport).map((p) => (
          <div key={p[0]} className="progress-item">
            <div className="ins-type">{p[0]}</div>
            <div className="ins-premium">
              &#8377;
              {' '}
              {p[1].map((e) => e.premium).reduce((a, b) => a + b)}
            </div>
            <div className="ins-count">
              Quantity:
              {' '}
              <span>{p[1].length}</span>
            </div>
          </div>
        ))}
      </div>
    </>
  );
};

export default Progress;
