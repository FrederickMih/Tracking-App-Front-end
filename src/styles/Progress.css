.progress__container {
  height: 81%;
  overflow: auto;
}

.progress__container h2 {
  text-align: center;
  padding-top: 1rem;
}
