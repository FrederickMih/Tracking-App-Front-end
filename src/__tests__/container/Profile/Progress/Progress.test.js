import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../../Redux/configureStore';
import Progress from '../../../../components/container/Progress';

const store = configureAppStore();

describe('Progress', () => {
  it('should match the snapshot of Progress component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Progress />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
