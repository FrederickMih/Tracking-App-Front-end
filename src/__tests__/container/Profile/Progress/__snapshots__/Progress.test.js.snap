// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should match the snapshot of Progress component 1`] = `
Array [
  <div
    className="headline"
  >
    Progress Report
    <div
      className="bkbar"
    >
      <a
        href="/add"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
      </a>
    </div>
  </div>,
  <div
    className="progress-stats"
  >
    <div>
      <div
        className="MuiBox-root MuiBox-root-1"
      >
        <div
          aria-valuenow={100}
          className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-determinate"
          role="progressbar"
          style={
            Object {
              "height": 80,
              "transform": "rotate(-90deg)",
              "width": 80,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate"
              cx={44}
              cy={44}
              fill="none"
              r={18.5}
              strokeWidth={7}
              style={
                Object {
                  "strokeDasharray": "116.239",
                  "strokeDashoffset": "0.000px",
                }
              }
            />
          </svg>
        </div>
        <div
          className="MuiBox-root MuiBox-root-2"
        >
          <div
            className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
          >
            5100%
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="progress-items"
  />,
]
`;
